### PR TITLE
[Foundation] Update hashing for bridged NSError protocols

### PR DESCRIPTION
**Pull request by lorentey**
Protocols that wish to provide a default implementation of hashing should do so by providing a `hash(into:)` implementation. `hashValue` has been deprecated as a `Hashable` requirement in SE-0206. 

Protocol extensions that only implement hashValue force `hashValue`-based synthesis on conforming types by default, which will probably produce deprecation warnings soon.

Add missing Hashable requirements to `_BridgedNSError` and `_BridgedStoredNSError`.

It is important to add these extensions now, since the presence/absence of defaulted Hashable requirements can cause [ABI headaches](https://github.com/apple/swift/pull/20705) down the road for conforming types, and these protocols are ~~used by ClangImporter~~ involved in automatic conformance synthesis.

rdar://problem/46496484

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci test

**Review by jrose-apple — APPROVED** [review]

File: stdlib/public/SDK/Foundation/NSError.swift [hunk 1]
```diff
@@ -406,7 +406,15 @@ extension _BridgedNSError where Self.RawValue: FixedWidthInteger {
     self.init(rawValue: RawValue(_bridgedNSError.code))
   }
 
+  public func hash(into hasher: inout Hasher) {
+    hasher.combine(_code)
+  }
+
   public var hashValue: Int { return _code }
```
Comment (DougGregor) [inline stdlib/public/SDK/Foundation/NSError.swift]: Should you remove `hashValue` and let it pick up the default implementation?

File: stdlib/public/SDK/Foundation/NSError.swift [hunk 2]
```diff
@@ -480,9 +488,17 @@ public extension _BridgedStoredNSError {
...
 /// Implementation of Hashable for all _BridgedStoredNSErrors.
-public extension _BridgedStoredNSError {
-  var hashValue: Int {
-    return _nsError.hashValue
+extension _BridgedStoredNSError {
+  public func hash(into hasher: inout Hasher) {
+    hasher.combine(_nsError)
+  }
+
+  public var hashValue: Int {
```
Comment (DougGregor) [inline stdlib/public/SDK/Foundation/NSError.swift]: Same question

**Review by DougGregor — APPROVED** [review]

File: stdlib/public/SDK/Foundation/NSError.swift (hunk 1 above)
Comment (lorentey) [reply stdlib/public/SDK/Foundation/NSError.swift]: That's a good point; I don't see any reason why we wouldn't want that. (In fact, the same applies to `_rawHashValue`.) I'll remove these.

**Comment by lorentey** [comment]
@swift-ci please test

**Comment by lorentey** [comment]
@swift-ci please test

**Pull request merged**